src/cmd/services/m3dbnode/config/bootstrap.go

```diff
@@ -70,18 +74,46 @@ type BootstrapConfiguration struct {
 type BootstrapFilesystemConfiguration struct {
 	// NumProcessorsPerCPU is the number of processors per CPU.
 	NumProcessorsPerCPU float64 `yaml:"numProcessorsPerCPU" validate:"min=0.0"`
+
+	// Migrations configuration
+	Migrations *BootstrapMigrations `yaml:"migrations"`
 }
 
 func (c BootstrapFilesystemConfiguration) numCPUs() int {
 	return int(math.Ceil(float64(c.NumProcessorsPerCPU * float64(runtime.NumCPU()))))
 }
 
+func (c BootstrapFilesystemConfiguration) migrations() BootstrapMigrations {
+	if cfg := c.Migrations; cfg != nil {
+		return *cfg
+	}
+	return BootstrapMigrations{Concurrency: defaultMigrationConcurrency}
+}
+
 func newDefaultBootstrapFilesystemConfiguration() BootstrapFilesystemConfiguration {
 	return BootstrapFilesystemConfiguration{
 		NumProcessorsPerCPU: defaultNumProcessorsPerCPU,
+		Migrations:          &BootstrapMigrations{},
 	}
 }
 
+// BootstrapMigrations specifies configuration for data migrations during bootstrapping
+type BootstrapMigrations struct {
```

Comment:
nit: `BootstrapMigration` or `BootstrapMigrationConfiguration` would work better here

Reply:
Given all other types here have a "...Configuration" suffix, I'd suggest making this "BootstrapMigrationConfiguration".